backend/src/agent/persistence/worker/metrics_reporter_test.go

```diff
@@ -61,18 +87,50 @@ func TestReportMetrics_NoArtifact_NoOP(t *testing.T) {
 			Namespace: "MY_NAMESPACE",
 			Name:      "MY_NAME",
 			UID:       types.UID("run-1"),
+			Labels:    map[string]string{util.LabelKeyWorkflowRunId: "run-1"},
+		},
+		Status: workflowapi.WorkflowStatus{
+			Nodes: map[string]workflowapi.NodeStatus{
+				"node-1": workflowapi.NodeStatus{
+					ID:    "node-1",
+					Phase: workflowapi.NodeSucceeded,
+				},
+			},
+		},
+	})
+	err := reporter.ReportMetrics(workflow)
+	assert.Nil(t, err)
+	assert.Nil(t, pipelineFake.GetReadArtifactRequest())
```

Comment:
Note, this is the reproducing test case. Without new changes, this will fail because persistence agent is still sending an artifact request in this case.